### DISCUSSION TITLE
Problem installing llvm when cmake and ninja are already installed

### DISCUSSION THREAD

**Discussion by perdrix52** [Homebrew/discussions#5749] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: cc0b5d6d05d1f3af5b2f53e9fbb1e7bfa0c332ab
Last commit: 3 days ago
Core tap JSON: 20 Nov 10:41 UTC
Core cask tap JSON: 20 Nov 10:00 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 3.3.6 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: quad-core 64-bit skylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 12.7.6-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
david@Mac-x64 ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  cmake

Warning: You are using macOS 12.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.

david@Mac-x64 ~ %
```

### Description of issue

I have already installed CMake using the standard .dmg file followed by:
```
sudo "/Applications/CMake.app/Contents/bin/cmake-gui" --install=/usr/local/bin
```
to install it to /usr/local/bin

I had also already installed ninja using **brew install ninja**

When I went to install llvm, brew tried to install both of these again :(  The result was:
```
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/ccmake
Target /usr/local/bin/ccmake
already exists. You may want to remove it:
  rm '/usr/local/bin/ccmake'

To force the link and overwrite all conflicting files:
  brew link --overwrite cmake

To list all files that would be deleted:
  brew link --overwrite cmake --dry-run

Possible conflicting files are:
/usr/local/bin/ccmake -> /Applications/CMake.app/Contents/bin/ccmake
/usr/local/bin/cmake -> /Applications/CMake.app/Contents/bin/cmake
/usr/local/bin/cpack -> /Applications/CMake.app/Contents/bin/cpack
/usr/local/bin/ctest -> /Applications/CMake.app/Contents/bin/ctest
Error: Could not symlink share/emacs/site-lisp/ninja/ninja-mode.el
Target /usr/local/share/emacs/site-lisp/ninja/ninja-mode.el
is a symlink belonging to ninja. You can unlink it:
  brew unlink ninja

To force the link and overwrite all conflicting files:
  brew link --overwrite ninja

To list all files that would be deleted:
  brew link --overwrite ninja --dry-run
```

Did the llvm install succeed or not - what should I do now
David

**Comment by p-linnane** [comment]
You're on an unsupported version of macOS. This means you can't use our prebuilt binaries and will have to compile everything yourself. The issue you're seeing is `brew` trying to install the `cmake` formula to build `llvm`. The `cmake` formula installed but did not link into your prefix since you already have a non-brewed `cmake` install.

**Comment by perdrix52** [comment]
But but why should I not have installed cmake the OFFICIAL way and why was the ninja stuff also failing (and that was installed using brew).

You should not just barf because cmake was already installed. :(
You may say unsupported level of macOS - - we *need* to build on old releases sometimes.

**Reply by SMillerDev** [reply]
Not sure what makes something official or not in your opinion. But if you didn't install it with `brew install cmake` homebrew won't know about it and will install its own version on top of it.

**Reply by p-linnane** [reply]
>But but why should I not have installed cmake the OFFICIAL way and why was the ninja stuff also failing (and that was installed using brew).

Because Homebrew uses its own dependencies.

>You should not just barf because cmake was already installed. :(

This is exactly what we should be doing. Homebrew uses its own cmake formula, not whatever random one is installed.

>You may say unsupported level of macOS - - we need to build on old releases sometimes.

This isn't an opinion. Apple no longer supports macOS Monterey, and neither do we. `brew` should still largely work, but we aren't building bottles for Monterey anymore. You're on your own there.

**Comment by perdrix52** [comment]
Official as in that's how you're supposed to install cmake according to Kitware - thats about as official as it gets
D.

**Reply by SMillerDev** [reply]
Then why didn't you install ninja and llvm from their upstream downloads? Brew isn't the official source for those either.

**Comment by perdrix52** [comment]
Kitware's build of cmake isn't a "random one"

**Reply by SMillerDev** [reply]
It's definitely a random install. You even changed the default install location to no longer be standard. 

It might be from an official source, but it's a random cmake install
